### PR TITLE
Create largest_square_area_in_matrix.py

### PR DESCRIPTION
**Pull request by kondekarshubham123**
### Describe your change:

Added my approach for following question

Question:
Given a binary matrix mat of size n * m, find out the maximum size square
sub-matrix with all 1s.

---
Example 1:

Input:
n = 2, m = 2
mat = [[1, 1],
       [1, 1]]

Output:
2

* [X] Add an algorithm?
* [X] Fix a bug or typo in an existing algorithm?
* [X] Documentation change?

### Checklist:
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [X] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [X] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [X] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: matrix/largest_square_area_in_matrix.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+bottom right corner is the cell with index (i,j) in the original matrix.
+
+Starting from index (0,0), for every 1 found in the original matrix,
+we update the value of the current element as
+
+dp_array(i,j)=dp_array(dp(i−1,j),dp_array(i−1,j−1),dp_array(i,j−1)) + 1.
+"""
+
+
+from typing import List
```
Comment (CaedenPH) [inline matrix/largest_square_area_in_matrix.py:46]:
```suggestion
```
You don't use this

File: matrix/largest_square_area_in_matrix.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        next_row = current_row
+
+    return largest_square_area
+
+
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
+    # print(largest_square_area_in_matrix_bottom_up(2, 2, [[1,1], [1,1]]))
```
Comment (CaedenPH) [inline matrix/largest_square_area_in_matrix.py:193]:
```suggestion
    print(largest_square_area_in_matrix_bottom_up(2, 2, [[1,1], [1,1]]))
```

**Review by CaedenPH — COMMENTED** [review]
Looks good - small changes

**Comment by kondekarshubham123** [comment]
Suggested small changes have been implemented,

Please review @cclauss , @CaedenPH

**Comment by cclauss** [comment]
Is there a function missing doctests?

Click `Details` to the right of the ❌ below to see misspelled words and long lines (max 88 chars per line).

**Comment by kondekarshubham123** [comment]
Actually the for recursive code, inside a parent function 
I haven't added the doctest, do I need to add?

**Comment by cclauss** [comment]
Cool.  Doctests do not run in functions that are nested under other functions so we can ignore the keeperbot on that one.

**Pull request merged**